avaframe/com1DFAPy/com1DFAtools.py

```diff
@@ -298,6 +323,160 @@ def computeTimeStep(cfg, particles, fields, dem, Ment, Cres, Tcpu):
+        print((forceY / mass)[ind])
+        print((forceZ / mass)[ind])
+        print(uNew[ind])
+        print(A)
+
+
+    #++++++++++++++UPDATE Particle Properties
+    # update mass required if entrainment
+    massNew = mass + force['dM']
+    particles['mTot'] = np.sum(massNew)
+    particles['x'] = xNew
+    particles['y'] = yNew
+    particles['s'] = particles['s'] + np.sqrt((xNew-xK)*(xNew-xK) + (yNew-yK)*(yNew-yK))
+    # make sure particle is on the mesh (recompute the z component)
+    particles, _ = geoTrans.projectOnRasterVect(dem, particles, interp='bilinear')
+
+    nx, ny, nz = getNormalArray(xNew, yNew, Nx, Ny, Nz, csz)
+    particles['m'] = massNew
+    # normal component of the velocity
+    uN = uxNew*nx + uyNew*ny + uzNew*nz
+    # print(nx, ny, nz)
+    # print(norm(ux, uy, uz), uN)
+    # remove normal component of the velocity
+    particles['ux'] = uxNew - uN * nx
+    particles['uy'] = uyNew - uN * ny
+    particles['uz'] = uzNew - uN * nz
+
+    #################################################################
+    # this is dangerous!!!!!!!!!!!!!!
+    ###############################################################
+    # remove particles that are not located on the mesh any more
+    particles = removeOutPart(cfg, particles, dem)
+
+
+    #+++++++++++++++++++++COMPUTE FLOW DEPTH
+    # plot flow depth computed with different interpolation methods
+    nSave = Tcpu['nSave']
+    dtSave = cfg.getfloat('dtSave')
+    if debugPlot and particles['t'] >= nSave * dtSave:
+        hNN = copy.deepcopy(particles['hNearestNearest'])
+        hNB = copy.deepcopy(particles['hNearestBilinear'])
+        hSPH = copy.deepcopy(particles['hSPH'])
+        hBN = copy.deepcopy(particles['hBilinearNearest'])
+        hBB = copy.deepcopy(particles['hBilinearBilinear'])
+        indexSort = np.argsort(hNN)
+        indexSortBB = np.argsort(hBB)
+
+        # print(np.mean(hSPH))
+        # fig, ax = plt.subplots(figsize=(2*figW, figH))
+        # ax.set_title('flow depth per particle')
+        # # ax.plot(hSPH[indexSort], 'b', linewidth=0.5, label='h from SPH')
+        # ax.plot(hBB[indexSort], 'y', linewidth=0.5, label='h from bilinear bilinear')
+        # # ax.plot(hBN[indexSort], 'g', linewidth=0.5, label='h from bilinear nearest')
+        # ax.plot(hNB[indexSort], 'r', linewidth=0.5, label='h from nearest bilinear')
+        # ax.plot(hNN[indexSort], 'k', linewidth=0.5, label='h from nearest nearest')
+        # plt.legend()
+        # plt.show()#block=False)
+        # plt.pause(1)
+        # plt.close()
+        ind = np.where(((particles['y']>905) & (particles['y']<1005)))
+        # fig2 = plt.figure()
+        # ax2 = fig2.add_subplot(111, projection='3d')
+        # ax2.scatter(particles['x'][ind], particles['y'][ind], hNN[ind], 'k')
+        # ax2.scatter(particles['x'][ind], particles['y'][ind], hBB[ind], 'b')
+        # ax2.set_xlim3d(0, 1000)
+        # ax2.set_ylim3d(950,1050)
+        # ax2.set_zlim3d(0,1000)
```

Comment:
Are all those comments still needed?

Reply:
removed the debugPlot from this feature's function